programmatic_migration.go

```diff
@@ -7,36 +7,53 @@ import "github.com/golang-migrate/migrate/v4/database"
 // the database driver as arguments.
 type ProgrammaticMigration func(migr *Migration, driver database.Driver) error
 
+// ProgrammaticMigrEntry configures how a programmatic migration should run.
+type ProgrammaticMigrEntry struct {
+	// RerunOnError determines the behavior if the programmatic migration
+	// errors. If set to `true` the migration logic will reset the database
+	// to the prior version, so the migration is retried on the next
+	// startup.
+	// If set to `false`, the database will be left in dirty state at the
+	// migration version of the programmatic migration.
+	RerunOnError bool
```

Comment:
can we rename this one to something more idiomatic, I first thought something is rerun, but its just rolled back, maybe `RollbackOnError` ?